packages/build/src/plugins_core/blobs_upload/utils.ts

```diff
@@ -0,0 +1,61 @@
+import fs from 'node:fs'
+import fsPromises from 'node:fs/promises'
+import path from 'node:path'
+
+import { fdir } from 'fdir'
+
+const metadataPrefix = '$'
+const metadataSuffix = '.json'
+
+export const getBlobsDir = function ({ buildDir, publishDir }) {
+  const blobsPath = '.netlify/blobs/deploy'
+  const absoluteBlobsPath = path.resolve(buildDir, publishDir, blobsPath)
+  const blobsDir = path.normalize(absoluteBlobsPath)
+  return blobsDir
+}
+
+export const anyBlobsToUpload = function ({ buildDir, publishDir }) {
+  const blobsDir = getBlobsDir({ buildDir, publishDir })
+  const { files } = new fdir().onlyCounts().crawl(blobsDir).sync()
```

Comment:
shouldn't we use the promise based method here? I think it will be way faster as it can leverage async I/O

```suggestion
  const { files } = await new fdir().onlyCounts().crawl(blobsDir).withPromise()
```

Reply:
Furthermore we are crawling the dir multiple times wouldn't make it sense to do this kind of cost-intensive operation once? 

I see you use it a second time on `getKeysToUpload`

Reply:
Updated to use the async version on 67684e2e3
I'm not too worried about doing the operation twice, the count should be pretty quick. If you're concerned we can memoize it but I'm less familiar with how to do that in this setup.